src/Raven.Server/Documents/Expiration/ExpirationStorage.cs

```diff
@@ -175,15 +178,18 @@ public record ExpiredDocumentsOptions
                                         expiredDocs.Add((clonedId, id));
                                     }
                                 }
-                            } while (multiIt.MoveNext() && expiredDocs.Count + count < options.AmountToTake);
+                            } while (multiIt.MoveNext() 
+                                     && count < options.AmountToTake 
+                                     && count < options.MaxItemsToProcess);
```

Comment:
`count < options.MaxItemsToProcess` isn't sufficient
in case of batched expiration (`options.AmountToTake` set lower than `MaxItemsToProcess`) we will exceed the limit

e.g. we set `batchSize=1024` and `maxItemsToProcess=1536`, and there are `2048 expired docs`, we would exceed max items limit in this case

we only use `count` here to track if the `MaxItemsToProcess` limit wasn't exceeded, but we need `currentTotalCount`

 https://github.com/ravendb/ravendb/pull/18223/files#r1521179121

```suggestion
                                     && currentTotalCount + count < options.MaxItemsToProcess);
```

Reply:
I am now initializing `count` to `totalCount` (instead 0), so I think it is the same, isn't it?

Reply:
Not really, because you'll start with a maxed out count, which may exceed `AmountToTake`

Reply:
done